model_analyzer/record/metrics_manager.py

```diff
@@ -160,10 +172,15 @@ def _start_monitors(self, cpu_only=False):
         """
 
         if not cpu_only:
-            self._dcgm_monitor = DCGMMonitor(
-                GPUDeviceFactory.get_analyzer_gpus(self._config.gpus),
-                self._config.monitoring_interval, self._dcgm_metrics)
-            self._dcgm_monitor.start_recording_metrics()
+            try:
+                self._dcgm_monitor = DCGMMonitor(
+                    self._gpus, self._config.monitoring_interval,
+                    self._dcgm_metrics)
+                self._check_triton_and_model_analyzer_gpus()
+                self._dcgm_monitor.start_recording_metrics()
+            except TritonModelAnalyzerException as e:
+                self._destroy_monitors()
+                raise TritonModelAnalyzerException(e)
```

Comment:
I think you can just use `raise` to reraise the exception.

Reply:
ah thank you!